### ISSUE TITLE
Possibility to control start of connection with hub

### ISSUE THREAD

**Issue by kobrynsky** [Omhet/redux-signalr#4]
Hi there!
Is there possibility to manage when connection with hub will start? 
On my project example: 
I have on the backend  (.NET Core WebApi):
1. AuthController - with one endpoint "Login" (it saves to cookie nickname, thats all i need)
2. GameHub - all of the signalr methods

I want to make connection with hub **after** user logs in. Why? Because if the connection is established before I save cookies, there are no cookies in GameHub methods.

**Comment by Omhet** [comment]
Hi!

Firstly, I would suggest you to try https://github.com/pofigizm/redux-dynamic-middlewares. With that one you can add redux-signalr in runtime when you need it.

If that doesn't help, plz, write here and we will try to come up with another solution. 

Thanks!

**Comment by kobrynsky** [comment]
Hi!
Thats great idea! I used it and 'connection flow" works how I want but whenever i want to use invoke in my redux methods I get error like this:
![image](https://user-images.githubusercontent.com/26911602/94781043-95b66300-03c9-11eb-8df4-db8d18df7b8e.png)

Seems like invoke is undefined (but getState and dispatch is not)

Do you know how I can solve this?
Thanks for your help, I really appreciate that :)

**Comment by Omhet** [comment]
Hm, that's interesting...

Well, I will try to investigate on this in my spare time, but I think it may take around a week. So, if you are interested, you can also investigate further why the 'invoke' is undefined.

**Comment by Omhet** [comment]
Or actually, I think there is one another way.
 I could add an optional param to start a connection immediately, and if this param is false, then the user of redux-signalr can decide himself when he wants to start the connection. I think it may work and no dynamic middlewares will be needed.

**Comment by Omhet** [comment]
So, I've tried my guess.

Here is the possible working code:

Create signalMiddleware with shouldConnectionStartImmediately set to false.
```typescript  
export const connection = new HubConnectionBuilder()
  .configureLogging(LogLevel.Debug)
  .withUrl("https://0.0.0.0:5001/testHub", {
    skipNegotiation: true,
    transport: HttpTransportType.WebSockets,
  })
  .build();

const signal = signalMiddleware({
  callbacks,
  connection,
  shouldConnectionStartImmediately: false
});
```

Then, import the 'connection' in the place you want and start it if it's not already.
Here is an example with a simple Button container:
```typescript
import { connection } from "../redux/helpers/createSignalMiddleware";

const StartConnectionButton: FunctionComponent = () => {
  const handleClick = useCallback(() => {
    if (connection.state !== HubConnectionState.Connected) {
      connection
        .start()
        .then(() => console.log("Connection started"))
        .catch((err) => console.error(err.toString()));
    }
  }, []);

  return <Button onClick={handleClick}>Start Connection</Button>;
};

export default StartConnectionButton;
```

**Comment by Omhet** [comment]
@blc132 

So, how do you like the possible solution? Do you think it will fit your needs?

**Comment by Omhet** [comment]
@blc132 

I have published an alpha version with this feature added.

Please, remove redux-signalr and install redux-signalr@alpha. Try to use it with an example in my previous comments. 
Write me back then 😊

**Comment by Omhet** [comment]
Now, it's published in 1.1.0, I thought that it's quite a good feature to have.

Also, I've already published 1.2.0 where I add a possibility to call dispatch, getState and invoke methods from callbacks.
Types were improved too. You can check out an updated README, if you are interested :)

**Comment by kobrynsky** [comment]
@Omhet 
Because of deadlines I was forced to use "temporary workaround", but I am pretty sure your feature would work in my case.

Thank you very much, I am pretty sure that i will use new version of your package on my next projects!

Greetings 😄
